docs/sources/developers/plugins/data-sources-for-logs.md

```diff
@@ -0,0 +1,144 @@
++++
+title = "Build a logs data source"
+type = "docs"
++++
+
+# Build a logs data source
+
+This guide explains how to build a logs data source plugin.
+
+Data sources in Grafana supports both metrics and log data. The steps to build a logs data source are largely the same as for a metrics data source. This guide assumes that you're already familiar with how to [Build a data source plugin]({{< relref "../../../../../tutorials/build-a-data-source-plugin.md" >}}) for metrics.
+
+
+1. **Enable logs support**.
+
+   Let Grafana know that your data source plugin can return log data, by adding `"logs": true` to the [plugin.json]({{< relref "metadata.md" >}}) file.
+
+   ```json
+   {
+     "logs": true
+   }
+   ```
+
+1. **Construct the log data**.
+
+   Just like for metrics data, Grafana expects your plugin to return log data as a [data frame]({{< relref "data-frames.md" >}}). Grafana considers any data frame with at least one time field and one text field to be log data.
+
+   ```ts
+   return new MutableDataFrame({
+     refId: query.refId,
+     fields: [
+       {
+         name: 'time',
+         type: FieldType.time,
+         values: [from, to]
+       },
+       {
+         name: 'content',
+         type: FieldType.string,
+         values: ['user registered', 'user logged in']
+       },
+     ],
+   });
+   ```
+
+That's all you need to start returning log data from your data source. Go ahead and try it out in [Explore]({{< relref "../../features/explore.md#explore" >}}), or by adding a [Logs panel]({{< relref "../../features/panels/logs.md#logs-panel" >}}).
+
+Congratulations, you just wrote your first logs data source plugin! Next, let's look at a couple of features that can improve the experience for the user.
+
+
+## Labels
```

Comment:
I suggest changing this heading and the next to "Verb nouns" or "Verb a noun" to maintain consistency with the earlier headings/steps.